### PR TITLE
Fix lplist issues

### PR DESCRIPTION
**Pull request by iverc**
- DO NOT MERGE

### CODE REVIEW

**Comment by timlind** [comment]
Marked as draft so we don't merge into develop yet, just needs to be deployed to tesnet-042 for now. Should maybe be merged into a common branch with https://github.com/Sifchain/sifnode/pull/1672

File: x/clp/keeper/liquidityprovider.go [hunk 1]
```diff
@@ -71,18 +73,26 @@ func (k Keeper) DestroyLiquidityProvider(ctx sdk.Context, symbol string, lpAddre
...
-	var lpList []types.LiquidityProvider
+func (k Keeper) GetLiquidityProvidersForAsset(ctx sdk.Context, asset types.Asset) []*types.LiquidityProvider {
+	var lpList []*types.LiquidityProvider
 	iterator := k.GetLiquidityProviderIterator(ctx)
 	defer iterator.Close()
 	for ; iterator.Valid(); iterator.Next() {
 		var lp types.LiquidityProvider
 		bytesValue := iterator.Value()
+		if !(len(bytesValue) > 0) {
+			panic("GetLiquidityProvidersForAsset ==>> len(bytesValue) <= 0")
```
Comment (marshsif) [inline x/clp/keeper/liquidityprovider.go:84]: Should we be panicking here? What if there are no LPs for a given asset, shouldn't that just return an error?

Comment (iverc) [reply x/clp/keeper/liquidityprovider.go:84]: it is for testing purposes, the logic should be updated after fixing the issue

**Review by marshsif — APPROVED** [review]

File: x/clp/keeper/liquidityprovider.go [hunk 2]
```diff
@@ -71,18 +73,26 @@ func (k Keeper) DestroyLiquidityProvider(ctx sdk.Context, symbol string, lpAddre
...
+		}
 		k.cdc.MustUnmarshalBinaryBare(bytesValue, &lp)
+		if lp.Asset == nil {
+			panic(fmt.Sprintf("GetLiquidityProvidersForAsset ==>> lp.Asset == nil, len(bytesValue): %d", len(bytesValue)))
```
Comment (intl-man) [inline x/clp/keeper/liquidityprovider.go:88]: @iverc Should we be using panic like this? not log and return?

File: x/clp/keeper/liquidityprovider.go (hunk 1 above)
Comment (intl-man) [inline x/clp/keeper/liquidityprovider.go:84]: @iverc Should we be using panic like this? not log and return?

**Pull request closed without merging**